sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -272,11 +272,11 @@ private[joins] final class UnsafeHashedRelation(
     while (i < nKeys) {
       val keySize = in.readInt()
       val valuesSize = in.readInt()
-      if (keySize > keyBuffer.size) {
+      if (keySize > keyBuffer.length) {
```

Comment:
unrelated change, but didn't want to submit a patch for 2 lines

cc @davies: we should call .length on arrays, instead of size, since .size actually creates a wrapper object, and then calls size on that ...